lib/Recur/RRuleIterator.php

```diff
@@ -349,7 +392,7 @@ protected function nextDaily(): void
     protected function nextWeekly(): void
     {
         if (!$this->byHour && !$this->byDay) {
-            $this->currentDate = $this->currentDate->modify('+'.$this->interval.' weeks');
+            $this->advanceTheDate('+'.$this->interval.' weeks');
 
             return;
         }
```

Comment:
This is a bit trickier. I will try have a look tonight. I just need to make sure that various BYDAY BYHOUR etc filters will skip along correctly while sometimes happening to hit a summer-time start day. There might be a few code paths to think about (or it might turn out to be easy!)

Reply:
I have done the scenario for WEEKLY BYDAY when the recurrence falls on Sunday at 0230.

It is trickier for: `FREQ=WEEKLY;INTERVAL=2;BYDAY=SA,SU;WKST=MO;BYHOUR=2,14`

The RRULE says to schedule at 0200 and 1400 on each Saturday and Sunday.
On the Sunday of summer-time transition there is no 0200. The code current skips scheduling anything on the Sunday morning of summer-time transition. Future recurrences on later weekends are correctly scheduled at 0200 and 1400.

The "bug" in this case is not that the scheduled time gets locked forward to 0300, but that the recurrence on the summer-time transition day is not scheduled at all.

The RFC does not say anything about what is required if one of the hours mentioned in BYHOUR does not exist on the day in question.

Reply:
@schreven I have handled all the cases that I could see where `nextWeekly` `netMonthly` or `nextYearly` have BYDAY or similar specified and the resulting day in the week, month or year might fall on a summer-time start date.
There are test cases and code fixes for all that.

Reply:
Got it, thanks for the details

> The "bug" in this case is not that the scheduled time gets locked forward to 0300, but that the recurrence on the summer-time transition day is not scheduled at all

Indeed that appears like a separate question that is out of scope

Looks good to me overall, with all the tests it's convincing that it behaves correctly